api/pkg/service/resource/resource.go

```diff
@@ -143,6 +139,28 @@ func (s *service) ByTypeName(ctx context.Context, p *resource.ByTypeNamePayload)
 	return s.resourcesForQuery(q)
 }
 
+// Finds a resource information using resource id
+func (s *service) ByResourceID(ctx context.Context, p *resource.ByResourceIDPayload) (res *resource.Resource, err error) {
+	res = &resource.Resource{}
+
+	var one model.Resource
+
+	q := s.db.Scopes(withResourceDetails,
```

Comment:
could you please move `q` at start and `res` below where it is used? check other methods.